### PR TITLE
[feature](alter) support rename column for table with unique column id

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-TABLE-RENAME.md [hunk 1]
```diff
@@ -66,6 +66,18 @@ grammar:
...
+
+grammar:
+
+```sql
+RENAME COLUMN old_column_name new_column_name;
+```
+
+Notice:
+- Currently only tables with column unique id are supported
+
```
Comment (dataroaring) [inline docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-TABLE-RENAME.md:79]: Currently only tables with column unique id are supported, which are created with property 'light_schema_change'.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -4030,12 +4031,113 @@ public void replayRenamePartition(TableInfo tableInfo) throws MetaNotFoundExcept
...
+
+        if (colName.equalsIgnoreCase(newColName)) {
+            throw new DdlException("Same column name");
+        }
+
+        Map<Long, MaterializedIndexMeta> indexIdToMeta = table.getIndexIdToMeta();
+        for (Map.Entry<Long, MaterializedIndexMeta> entry : indexIdToMeta.entrySet()) {
+            // rename column is not implemented for table without column unique id.
+            if (entry.getValue().getMaxColUniqueId() < 0) {
+                throw new DdlException("not implemented for table without column unique id");
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:4048]: same as before, users do not know what is column unique id, they know property in create table statement.

**Review by dataroaring — CHANGES_REQUESTED** [review]
We should change meta version and add tests for mv.

**Comment by liaoxin01** [comment]
> We should change meta version and add tests for mv.

mv test has been added.

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 41.28 seconds
 load time: 596 seconds
 storage size: 17154743846 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221020222044_clickbench_pr_31967.html

**Pull request merged**